### PR TITLE
Fix ImageAnnotationRef and clean up

### PR DESCRIPTION
**Pull request by bramalingam**
@melissalinkert , @sbesson 

ref: https://trello.com/c/WgZKOpA4/20-clients-support-support-map-annotations-from-ome-tiff

### CODE REVIEW

File: components/formats-gpl/utils/writeMapAnnotationsExample.java [hunk 1]
```diff
@@ -1,16 +1,3 @@
...
-import loci.formats.ome.OMEXMLMetadata;
-import loci.formats.services.OMEXMLServiceImpl;
-import loci.formats.services.OMEXMLService;
-import loci.common.services.ServiceFactory;
-import loci.formats.meta.IMetadata;
-
-
 /*
  *------------------------------------------------------------------------------
  *  Copyright (C) 2014 University of Dundee. All rights reserved.
```
Comment (sbesson) [inline components/formats-gpl/utils/writeMapAnnotationsExample.java]: Fix the copyright year as well?

File: components/formats-gpl/utils/writeMapAnnotationsExample.java [hunk 2]
```diff
@@ -84,6 +86,17 @@ public static void main(String[] args) throws Exception{
         metadata.setMapAnnotationAnnotator("Example Map Annotation", mapAnnotationIndex);
         metadata.setMapAnnotationDescription("Example Description", mapAnnotationIndex);
         metadata.setMapAnnotationNamespace("Example NameSpace", mapAnnotationIndex);
+        metadata.setImageAnnotationRef("MapAnnotation",0, annotationRefIndex);
```
Comment (melissalinkert) [inline components/formats-gpl/utils/writeMapAnnotationsExample.java]: The string passed in here needs to match what was passed to ```setMapAnnotationID(...)```.

File: components/formats-gpl/utils/writeMapAnnotationsExample.java [hunk 3]
```diff
@@ -84,6 +86,17 @@ public static void main(String[] args) throws Exception{
...
+
+        mapAnnotationIndex = 1;
+        annotationRefIndex = 1;
+        String mapAnnotationID = MetadataTools.createLSID("MapAnnotation", 0, mapAnnotationIndex);
+        metadata.setMapAnnotationID(mapAnnotationID, mapAnnotationIndex);
+        metadata.setMapAnnotationValue(mapList, mapAnnotationIndex);
+        metadata.setMapAnnotationAnnotator("Example Map Annotation 1", mapAnnotationIndex);
+        metadata.setMapAnnotationDescription("Example Description 1", mapAnnotationIndex);
+        metadata.setMapAnnotationNamespace("Example NameSpace 1", mapAnnotationIndex);
+        metadata.setImageAnnotationRef("MapAnnotation 1",0, annotationRefIndex);
```
Comment (melissalinkert) [inline components/formats-gpl/utils/writeMapAnnotationsExample.java]: Same as above - ```MapAnnotation 1``` needs to be replaced with the ID passed to ```setMapAnnotationID(...)```.

**Comment by sbesson** [comment]
```
  [javac] warning: [options] bootstrap class path not set in conjunction with -source 1.6
    [javac] /home/travis/build/openmicroscopy/bioformats/components/formats-gpl/utils/writeMapAnnotationsExample.java:93: error: variable mapAnnotationID is already defined in method main(String[])
    [javac]         String mapAnnotationID = MetadataTools.createLSID("MapAnnotation", 0, mapAnnotationIndex);
```

**Comment by sbesson** [comment]
Tested the example file from this branch to generate a new sample OME-TIFF file. Checked using Bio-Formats command-line tools and by importing into OMERO that the Image is linked to two map annotations. Merging.

**Comment by bramalingam** [comment]
--rebased-to #2137

**Pull request merged**